File: conversions/django_models/construct.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Construct(AbstractElementModel):

    # Nature
    history = models.TextField(blank=True, null=True)
    understanding = models.TextField(blank=True, null=True)
    state = models.TextField(blank=True, null=True)
    start_date = models.PositiveIntegerField(blank=True, null=True)
    end_date = models.PositiveIntegerField(blank=True, null=True)
    founder = models.ForeignKey("Character", on_delete=models.SET_NULL, blank=True, null=True, related_name="construct_founder")
    organiser = models.ForeignKey("Institution", on_delete=models.SET_NULL, blank=True, null=True, related_name="construct_organiser")

    # Involves
    characters = models.ManyToManyField("Character", blank=True, related_name="construct_characters")
    objects = models.ManyToManyField("Object", blank=True, related_name="construct_objects")
    locations = models.ManyToManyField("Location", blank=True, related_name="construct_locations")
    species = models.ManyToManyField("Species", blank=True, related_name="construct_species")
    creatures = models.ManyToManyField("Creature", blank=True, related_name="construct_creatures")
    institutions = models.ManyToManyField("Institution", blank=True, related_name="construct_institutions")
    traits = models.ManyToManyField("Trait", blank=True, related_name="construct_traits")
    collectives = models.ManyToManyField("Collective", blank=True, related_name="construct_collectives")
    territories = models.ManyToManyField("Territory", blank=True, related_name="construct_territories")
    abilities = models.ManyToManyField("Ability", blank=True, related_name="construct_abilities")
    phenomena = models.ManyToManyField("Phenomenon", blank=True, related_name="construct_phenomena")
    languages = models.ManyToManyField("Language", blank=True, related_name="construct_languages")
    families = models.ManyToManyField("Family", blank=True, related_name="construct_families")
    relations = models.ManyToManyField("Relation", blank=True, related_name="construct_relations")
    titles = models.ManyToManyField("Title", blank=True, related_name="construct_titles")
    constructs = models.ManyToManyField("Construct", blank=True, related_name="construct_constructs")

    def __str__(self):
        return self.name

File: conversions/csharp_unity_json_properties/Construct.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Construct : Element
{
    [JsonProperty("history"), TextAttribute("")]
    public string history;
    [JsonProperty("understanding"), TextAttribute("")]
    public string understanding;
    [JsonProperty("state"), TextAttribute("")]
    public string state;
    [JsonProperty("start_date"), Integer(0)]
    public int startDate;
    [JsonProperty("end_date"), Integer(0)]
    public int endDate;
    [JsonProperty("founder"), ReferenceAttribute(typeof(Character))]
    public string founder;
    [JsonProperty("organiser"), ReferenceAttribute(typeof(Institution))]
    public string organiser;
    [JsonProperty("characters"), ReferenceAttribute(typeof(Character), true)]
    public string characters;
    [JsonProperty("objects"), ReferenceAttribute(typeof(Object), true)]
    public string objects;
    [JsonProperty("locations"), ReferenceAttribute(typeof(Location), true)]
    public string locations;
    [JsonProperty("species"), ReferenceAttribute(typeof(Species), true)]
    public string species;
    [JsonProperty("creatures"), ReferenceAttribute(typeof(Creature), true)]
    public string creatures;
    [JsonProperty("institutions"), ReferenceAttribute(typeof(Institution), true)]
    public string institutions;
    [JsonProperty("traits"), ReferenceAttribute(typeof(Trait), true)]
    public string traits;
    [JsonProperty("collectives"), ReferenceAttribute(typeof(Collective), true)]
    public string collectives;
    [JsonProperty("territories"), ReferenceAttribute(typeof(Territory), true)]
    public string territories;
    [JsonProperty("abilities"), ReferenceAttribute(typeof(Ability), true)]
    public string abilities;
    [JsonProperty("phenomena"), ReferenceAttribute(typeof(Phenomenon), true)]
    public string phenomena;
    [JsonProperty("languages"), ReferenceAttribute(typeof(Language), true)]
    public string languages;
    [JsonProperty("families"), ReferenceAttribute(typeof(Family), true)]
    public string families;
    [JsonProperty("relations"), ReferenceAttribute(typeof(Relation), true)]
    public string relations;
    [JsonProperty("titles"), ReferenceAttribute(typeof(Title), true)]
    public string titles;
    [JsonProperty("constructs"), ReferenceAttribute(typeof(Construct), true)]
    public string constructs;
}
